packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -201,98 +201,98 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
-        let buffer;
+    public async readBlobCore(blobId: string): Promise<IBlob | ArrayBuffer> {
+        let blob = this.blobCache.get(blobId);
         // Reset the timer on attempted cache read
         this.scheduleClearBlobsCache();
+
         if (blob === undefined) {
             this.checkAttachmentGETUrl();
 
-            const response = await getWithRetryForTokenRefresh(async (options) => {
+            blob = await getWithRetryForTokenRefresh(async (options) => {
                 const storageToken = await this.getStorageToken(options, "GetBlob");
-
                 const unAuthedUrl = `${this.attachmentGETUrl}/${encodeURIComponent(blobId)}/content`;
                 const { url, headers } = getUrlAndHeadersWithAuth(unAuthedUrl, storageToken);
 
                 return PerformanceEvent.timedExecAsync(
                     this.logger,
                     {
-                        eventName: "readBlob",
+                        eventName: "readDataBlob",
                         headers: Object.keys(headers).length !== 0 ? true : undefined,
                         waitQueueLength: this.epochTracker.rateLimiter.waitQueueLength,
                     },
-                    async () => this.epochTracker.fetchResponse(url, { headers }, FetchType.blob),
+                    async (event) => {
+                        const res = await this.epochTracker.fetchResponse(url, { headers }, FetchType.blob);
+                        const blobContent = await res.arrayBuffer();
+                        event.end({ size: blobContent.byteLength });
+                        return blobContent;
+                    },
                 );
             });
-            buffer = await response.arrayBuffer();
+            this.blobCache.set(blobId, blob);
         }
 
         if (!this.attributesBlobHandles.has(blobId)) {
-            if (blob) {
-                if (outputFormat === blob.encoding || (outputFormat === "string" && blob.encoding === undefined))  {
-                    return blob.content;
-                } else if (outputFormat === "base64") {
-                    return fromUtf8ToBase64(blob.content);
-                } else {
-                    return fromBase64ToUtf8(blob.content);
-                }
-            }
-
-            return IsoBuffer.from(buffer).toString(outputFormat === "base64" ? "base64" : "utf8");
+            return blob;
         }
-
         // ODSP document ids are random guids (different per session)
         // fix the branch name in attributes
         // this prevents issues when generating summaries
         let documentAttributes: api.IDocumentAttributes;
-        if (blob) {
-            documentAttributes = JSON.parse(blob.encoding === "base64" ? fromBase64ToUtf8(blob.content) : blob.content);
+        if (blob instanceof ArrayBuffer) {
+            documentAttributes = JSON.parse(IsoBuffer.from(blob).toString("utf8"));
         } else {
```

Comment:
Maybe I'm not correctly following the code here.  Shouldn't `blob instanceof ArrayBuffer` always be true here? It seems like we either get it from the cache (which is in-memory), or we get it from the `.arrayBuffer()` on the response now.

How can it be that it is not an ArrayBuffer? And more specifically, a string?

Reply:
Nevermind. I see now, it is below in case of documentAttributes, then we get `IBlob` (not string).